app/components/channel-nav/template.hbs

```diff
@@ -2,15 +2,18 @@
   <li>
     <h2>{{#link-to "space" space}}{{space.name}}{{/link-to}}</h2>
     <ul>
-    {{#each space.sortedChannels as |channel|}}
-      <li class="{{if channel.connected "connected" "disconnected"}} {{channel.unreadMessagesClass}}">
-        {{#if channel.isUserChannel}}
-          {{#link-to "space.userChannel" space channel}}{{channel.name}}{{/link-to}}
-        {{else}}
-          {{#link-to "space.channel" space channel}}{{channel.name}}{{/link-to}}
-        {{/if}}
+      {{#each space.sortedChannels as |channel|}}
+        <li class="{{if channel.connected "connected" "disconnected"}} {{channel.unreadMessagesClass}}">
+          {{#if channel.isUserChannel}}
+            {{#link-to "space.userChannel" space channel}}{{channel.name}}{{/link-to}}
+          {{else}}
+            {{#link-to "space.channel" space channel}}{{channel.name}}{{/link-to}}
+          {{/if}}
+        </li>
+      {{/each}}
+      <li>
+        <a {{action "joinChannel" space}} class="join-channel">+</a>
```

Comment:
If you do 

```
<a {{action (action joinChannel space.id)}} class="join-channel">+</a>
```

instead, you can get rid of the whole action definition in the component itself.

Action closures FTW :)